File: java/Core/src/main/java/com/devexpress/logify/alert/core/logger/ILogger.java
package com.devexpress.logify.alert.core.logger;

import java.util.List;
import java.util.Map;

public interface ILogger {
    StringBuilder getBuilder();
    Map<String, Object> getData();

    void beginWriteObject(String name);
    void writeValue(String name, String text);
    void writeValue(String name, boolean value);
    void writeValue(String name, int value);
    void writeValue(String name, List arrayList);
    void endWriteObject(String name);

    void beginWriteArray(String name);
    void endWriteArray(String name);
}

File: dotnet/Logify.Alert.Core/Logger/ILogger.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace DevExpress.Logify.Core.Internal {
    public interface ILogger {
        TextWriter Writer { get; }
        Dictionary<string, object> Data { get; }
        void BeginWriteObject(string name);
        void EndWriteObject(string name);
        void BeginWriteArray(string name);
        void EndWriteArray(string name);
        void WriteValue(string name, string text);
        void WriteValue(string name, bool value);
        void WriteValue(string name, int value);
        void WriteValue(string name, Array array);
    }
}